src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -155,6 +155,8 @@ public function setAutoEtag()
      */
     public function setContentDisposition($disposition, $filename = '', $filenameFallback = '')
     {
+        $filename = (string) $filename;
+
         if ('' === $filename) {
```

Comment:
if ('' === ($filename ?? '')) {

Reply:
Why not simplier `if (!$filename)` ?

Or `if (empty($filename))` ?

Reply:
it wouldn't be the same, as `'0'` is false|empty

Reply:
(We could also assume that filename is necessarily a string, as phpdoc doesn't allow it to be null, is it right ?)

Reply:
I believe so yes

Reply:
Ok, I'll remove the typecast.